File: final_dataset_A/generated_code/claude/java/Send_chat_message_024_Vulnerable.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class Send_chat_message_024_Vulnerable {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String sendMessage(int senderId, int recipientId, String message) {
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {

            String query = "INSERT INTO chat_messages (sender_id, recipient_id, message) " +
                    "VALUES (" + senderId + ", " + recipientId + ", '" + message + "')";
            stmt.executeUpdate(query);
            return "Message sent successfully";
        } catch (SQLException e) {
            return "Failed to send message: " + e.getMessage();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Send_chat_message_024_Vulnerable.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def send_message(sender_id, recipient_id, message):
    """Let a user send a chat message to another user."""
    conn = get_connection()
    cursor = conn.cursor()

    query = "INSERT INTO chat_messages (sender_id, recipient_id, message) " \
            "VALUES (%s, %s, '%s')" % (sender_id, recipient_id, message)
    try:
        cursor.execute(query)
        conn.commit()
        return "Message sent successfully"
    except Exception as e:
        return "Failed to send message: " + str(e)
    finally:
        conn.close()
